tests/runtest.py

```diff
@@ -143,20 +169,81 @@
 os.environ["MIDDLEWARE_TEST_PASSWORD"] = passwd
 os.environ["SERVER_TYPE"] = "ENTERPRISE_HA" if ha else "STANDARD"
 
-ip_to_use = ip if not ha else os.environ['controller1_ip']
+ip_to_use = ip
+if ha and ip2:
+    domain = 'tn.ixsystems.com'
+    os.environ['controller1_ip'] = ip
+    os.environ['controller2_ip'] = ip2
 
-with client(host_ip=ip_to_use) as c:
-    interface = c.call(
-        'interface.query',
-        [['state.aliases.*.address', '=', socket.gethostbyname(ip_to_use)]],
-        {'get': True}
-    )['id']
+
+def get_ipinfo(ip_to_use):
+    iface = net = gate = ns1 = ns2 = None
+    with client(host_ip=ip_to_use) as c:
+        net_config = c.call('network.configuration.config')
+        ns1 = net_config.get('nameserver1')
+        ns2 = net_config.get('nameserver2')
+        _ip_to_use = socket.gethostbyname(ip_to_use)
+        for i in c.call('interface.query'):
+            for j in i['state']['aliases']:
+                if j.get('address') == _ip_to_use:
+                    iface = i['id']
+                    net = j['netmask']
+                    for k in c.call('route.system_routes'):
+                        if k.get('network') == '0.0.0.0' and k.get('interface') == i['id']:
+                            if k['gateway']:
+                                return iface, net, k['gateway'], ns1, ns2
+
+    return iface, net, gate, ns1, ns2
+
+interface, netmask, gateway, ns1, ns2 = get_ipinfo(ip_to_use)
+if not all((interface, netmask, gateway)):
+    print(f'Unable to determine interface ({interface!r}), netmask ({netmask!r}) and gateway ({gateway!r}) for {ip_to_use!r}')
+    exit()
+
+os.environ['USE_VIP'] = 'NO'
+if ha:
+    if vip:
+        os.environ['virtual_ip'] = vip
+    elif os.environ.get('virtual_ip'):
+        vip = os.environ['virtual_ip']
+    else:
+        for i in ip_interface(f'{ip}/{netmask}').network:
+            last_octet = int(i.compressed.split('.')[-1])
+            if last_octet < 15 or last_octet >= 250:
```

Comment:
Might need to update the DNS?
My HA VM in the new KVM uses 10.220.38.[6.7.8]

Reply:
Your VM is using those as DNS servers?

Reply:
Oh I see what you're saying, this method takes into consideration the subnet mask so it it'll iterate over every address. There won't be any collisions.